### PR TITLE
Test unboxed numerics (part of unboxed-types version 2)

### PR DESCRIPTION
**Pull request by goldfirere**
@lpw25 and I thought it might be good to write tests to accurately define the expected functionality of upcoming releases of unboxed types. There doesn't seem to be a good reason not to just check this into the repo now, but I'm happy to consider other ways of doing this.

### CODE REVIEW

**Comment by goldfirere** [comment]
Chatted about this with @ceastlund, who may enjoy taking a look.

File: testsuite/tests/typing-layouts/unboxed-ints.ml [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+
+let x = Uint64.(#3L + #4L)
+let x = Uint32.(#3l + #4l)
+let x = Unativeint.(#3n + #4n)
+
+[%%expect {|
+success
+|}]
+
+let bad = #3L + #4L  (* wrong + *)
```
Comment (antalsz) [inline testsuite/tests/typing-layouts/unboxed-ints.ml:230]:
```suggestion
(* wrong + *)
let bad = #3L + #4L
```

**Review by antalsz — COMMENTED** [review]
This basically looks good to me!  A few comments made inline

**Comment by goldfirere** [comment]
These tests appear to be well represented in the current flambda-backend. Closing

**Pull request closed without merging**